src/common/hist_util.cc

```diff
@@ -33,19 +33,19 @@ void HistCutMatrix::Init(DMatrix* p_fmat, uint32_t max_num_bins) {
     s.Init(info.num_row_, 1.0 / (max_num_bins * kFactor));
   }
 
-  dmlc::DataIter<RowBatch>* iter = p_fmat->RowIterator();
+  auto iter = p_fmat->RowIterator();
```

Comment:
This seems to be a lot of different opinions on this topic.

I am using intellisense that shows the type when you mouse over so I get all of the benefits with none of the drawbacks. I can see how this would be slightly annoying if you are using a plain text editor. Nonetheless the row iterator function always returns the same type so I'm not sure it's a big deal.

FYI I am planning to replace these dmlc iterators with c++11 range based for loops. They are safer and considerably less verbose.

Reply:
> I am using intellisense that shows the type when you mouse over so I get all of the benefits with none of the drawbacks

I wish I had Visual Studio on my macbook xP. Oh well, I guess I always can use `grep` to find out the return type of `RowIterator()`.